server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQTerminationSubscriber.java

```diff
@@ -0,0 +1,138 @@
+    private static final String QUEUE_NAME_PREFIX = "terminationSubscriber";
+    private static final String ROUTING_KEY = "terminationSubscriberRoutingKey";
+
+    private final JsonEventSerializer serializer;
+    private final Mono<Connection> connectionMono;
+    private final ReactorRabbitMQChannelPool channelPool;
+    private final String queueName;
+    private UnicastProcessor<OutboundMessage> sendQueue;
+    private DirectProcessor<Event> listener;
+    private Disposable sendQueueHandle;
+
+    @Inject
+    public RabbitMQTerminationSubscriber(SimpleConnectionPool simpleConnectionPool, JsonEventSerializer serializer) {
+        this.serializer = serializer;
+        this.connectionMono = simpleConnectionPool.getResilientConnection();
+        this.channelPool = new ReactorRabbitMQChannelPool(connectionMono, MAX_CHANNELS_NUMBER);
+        this.queueName = QUEUE_NAME_PREFIX + UUID.randomUUID().toString();
+    }
+
+    public void start() {
+        Sender sender = channelPool.createSender();
+
+        sender.declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)).block();
+        sender.declare(QueueSpecification.queue(queueName).durable(false).autoDelete(true)).block();
+        sender.bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, queueName)).block();
+        sendQueue = UnicastProcessor.create();
+        sendQueueHandle = sender
+            .send(sendQueue)
+            .subscribeOn(Schedulers.elastic())
+            .subscribe();
+
+        Receiver receiver = RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(connectionMono));
+        listener = DirectProcessor.create();
+        receiver
+            .consumeAutoAck(queueName)
+            .subscribeOn(Schedulers.elastic())
+            .concatMap(this::toEvent)
+            .subscribe(listener::onNext);
+    }
+
+    @Override
+    public void addEvent(Event event) {
+        try {
+            byte[] payload = serializer.serialize(event).getBytes(StandardCharsets.UTF_8);
+            AMQP.BasicProperties basicProperties = new AMQP.BasicProperties.Builder().build();
+            OutboundMessage message = new OutboundMessage(EXCHANGE_NAME, ROUTING_KEY, basicProperties, payload);
+            sendQueue.onNext(message);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Publisher<Event> listenEvents() {
+        return listener
+            .share();
+    }
+
+    private Mono<Event> toEvent(Delivery delivery) {
+        String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
+        try {
+            Event event = serializer.deserialize(message);
+            return Mono.just(event);
+        } catch (Exception e) {
+            LOGGER.error("Unable to deserialize '{}'", message, e);
+            return Mono.empty();
+        }
+    }
+
+    @Override
+    @PreDestroy
+    public void close() {
```

Comment:
`close` is not consistent with `start`

Reply:
We used the `Startable` and  `Closeable` interfaces.
What would you recommend ?

Reply:
you could have stop method that would be an alias for close (or the other way around)

Reply:
No, it would be more confusing than useful, at best we could have a `Manageable` interface with `open`/`close` or `start`/`stop`, but you can't expect a coherent naming in uncorrelated interfaces.

Maybe it's a good point for the inter-sprint.

Reply:
What is your suggestion to solve the immediate problem, then? start/close is not ok IMO.